crates/net/network/src/peers/manager.rs

```diff
@@ -403,22 +403,27 @@ impl PeersManager {
 
     /// Called for a newly discovered peer.
     ///
-    /// If the peer already exists, then the address will be updated. If the addresses differ, the
-    /// old address is returned
-    pub(crate) fn add_discovered_node(&mut self, peer_id: PeerId, addr: SocketAddr) {
+    /// If the peer already exists, then the address will be updated.
+    pub(crate) fn add_discovered_node(
+        &mut self,
+        peer_id: PeerId,
+        kind: PeerKind,
+        addr: SocketAddr,
+    ) {
```

Comment:
we can stick to the same pattern mentioned above here as well.

So we'll rename `add_discovered_node` to `add_peer_kind`, etc..